TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Service/ImportExport/NodeImportService.php

```diff
@@ -267,7 +273,7 @@ protected function parseElement(\XMLReader $xmlReader)
                 $this->nodeIdentifierStack[] = $xmlReader->getAttribute('identifier');
                 // update current path
                 $nodeName = $xmlReader->getAttribute('nodeName');
-                if ($nodeName !== '/') {
+                if ($nodeName !== '/' && $nodeName !== '') {
```

Comment:
So, if the node name is empty, we process it as usual but do not add a name to the stack? That sounds wrong to me…

A node without a name should never occur in a site XML file, and if it does it should either throw an exception, be ignored and logged or get a generated name.